addons/project/static/src/services/project_task_recurrence.js

```diff
@@ -12,92 +12,43 @@ class ProjectTaskRecurrence {
         for (const task of tasks) {
             if (task.data.recurrence_id) {
-                taskIdsWithRecurrence.push(task.resId);
-                const recurrenceId = task.data.recurrence_id && task.data.recurrence_id[0];
-                if (!(recurrenceId in tasksPerRecurrence)) {
-                    tasksPerRecurrence[recurrenceId] = [task.resId];
-                    recurrenceIds.push(recurrenceId);
-                } else {
-                    tasksPerRecurrence[recurrenceId].push(task.resId);
-                }
+                recurrentTaskIds.push(task.resId);
+                hasRecurrentTasks = true;
             }
         }
-
-        if (!recurrenceIds.length) {
+        if (!hasRecurrentTasks) {
             callback();
             return;
         }
 
-        let allowContinue = false;
-        if (recurrenceIds.length === 1) {
-            const count = await this.orm.searchCount(
-                this.resModel,
-                [['recurrence_id', '=', recurrenceIds[0]]],
-            );
-            allowContinue = count != 1;
-        } else {
-            const taskReadGroup = await this.orm.readGroup(
-                this.resModel,
-                [['recurrence_id', 'in', recurrenceIds]],
-                ['recurrence_id'],
-                ['recurrence_id'],
-            );
-            allowContinue = true;
-            for (const res of taskReadGroup) {
-                const taskCount = tasksPerRecurrence[res.recurrence_id[0]].length;
-                if (taskCount === res.recurrence_id_count) {
-                    allowContinue = false;
-                    break;
-                }
-            }
-        }
-        let dialogBody;
-        if (tasks.length > 1) {
-            dialogBody = allowContinue
-                    ? this.env._t('It seems that some tasks are part of a recurrence.')
-                    : this.env._t('It seems that some tasks are part of a recurrence. At least one of them must be kept as a model to create the next occurences.');
-        } else {
-            dialogBody = allowContinue
-                    ? this.env._t('It seems that this task is part of a recurrence.')
-                    : this.env._t('It seems that this task is recurrent. Would you like to stop its recurrence?');
-        }
-
         const dialogProps = {
-            body: dialogBody,
-            confirm: async () => {
+            body: '',
+            confirm: () => {},
+            cancel: () => {},
+            addressRecurrence: async (target) => {
                 await this.orm.call(
                     this.resModel,
-                    'action_stop_recurrence',
-                    [taskIdsWithRecurrence],
+                    'action_address_recurrence',
+                    [recurrentTaskIds, target, mode],
                 );
                 callback();
             },
-            cancel: () => {},
+            modeLabel: mode.charAt(0).toUpperCase() + mode.slice(1),
```

Comment:
This will make the text not translatable

Reply:
Now that I adapted the code, everything that needs to be translated is in the template.